### PR TITLE
animal-object

### PR DESCRIPTION
**Pull request by maria-tsek**

### CODE REVIEW

File: jsarraymethods2/js-array-methods-2_animal-objects/index.js [hunk 1]
```diff
@@ -91,22 +94,41 @@ const indexOfAnimalWithNameLongerFive = null;
 // Hint: sort() mutates the original array, which is bad.
 // -> Use animals.slice().sort(...) to make a copy (and the tests work).
 
-const animalsSortedAlphabetically = null;
-
-const animalsSortedByWeightStartingWithLowest = null;
+const animalsSortedAlphabetically = animals.slice().sort((a, b) => {
```
Comment (reverbePrintemps) [inline jsarraymethods2/js-array-methods-2_animal-objects/index.js:97]: Why do we need to use the slice method?

File: jsarraymethods2/js-array-methods-2_animal-objects/index.js [hunk 2]
```diff
@@ -91,22 +94,41 @@ const indexOfAnimalWithNameLongerFive = null;
...
+  if (a.name < b.name) return -1;
+  if (a.name > b.name) return 1;
+  return 0;
```
Comment (reverbePrintemps) [inline jsarraymethods2/js-array-methods-2_animal-objects/index.js:100]: Nice and concise! 👍

File: jsarraymethods2/js-array-methods-2_animal-objects/index.js [hunk 3]
```diff
@@ -91,22 +94,41 @@ const indexOfAnimalWithNameLongerFive = null;
...
+const allAnimalsInEuropeWeighLessThanOnehundred = animals
+  .filter((animal) => animal.continents.includes("Europe"))
+  .every((animal) => animal.weight < 100);
 
 // Hint: filter + map + reduce
-const weightOfAllAnimalsInAfrica = null;
+const weightOfAllAnimalsInAfrica = animals
+  .filter((animal) => animal.continents.includes("Africa"))
+  .map((animal) => animal.weight)
+  .reduce((total, weight) => total + weight, 0);
```
Comment (reverbePrintemps) [inline jsarraymethods2/js-array-methods-2_animal-objects/index.js:123]: Love that you've used more explicit names for the reducer arguments here!

Comment (reverbePrintemps) [reply jsarraymethods2/js-array-methods-2_animal-objects/index.js:123]: ps: I _think_ that passing the initial value of `0` is unnecessary here.

**Review by reverbePrintemps — APPROVED** [review]
Looks great! Left a couple minor comments but also happy for this to be merged as is 👍

**Comment by reverbePrintemps** [comment]
ps: you didn't leave a description in the PR. That's 👎  though 😢

**Comment by maria-tsek** [comment]
> ps: you didn't leave a description in the PR. That's 👎 though 😢

don't cry you break my heart!

**Pull request merged**